src/main/java/Duke.java

```diff
@@ -1,10 +1,320 @@
+                "     ☹ OOPS!!! I'm sorry, but I don't know what that means :-(\n" +
+                "    ____________________________________________________________\n");
+    }
+
+    private static String loadExistingList(ArrayList<Task> tasks) {
+        checkSpecificFolderExist();
+        String dataListFilePath = "data/duke.txt";
+        try {
+            File dataListFile = checkFileExists(dataListFilePath);
+            readDataList(tasks, dataListFile);
+        } catch (IOException e) {
+            System.out.println("IOException error has occurred");
+        }
+        return dataListFilePath;
+    }
+
+    private static void checkSpecificFolderExist() {
+        File folder = new File("data/");
+        if (!folder.exists()) {
+            folder.mkdir();
+        }
+    }
+
+    private static File checkFileExists(String dataListFilePath) throws IOException {
+        File dataFile = new File(dataListFilePath);
+        if (!dataFile.exists()) {
+            dataFile.createNewFile();
+        }
+        return dataFile;
+    }
+
+    private static void readDataList(ArrayList<Task> tasks, File dataListFile) throws FileNotFoundException {
+        Scanner fileScanner = new Scanner(dataListFile);
+        while (fileScanner.hasNextLine()) {
+            String currentLine = fileScanner.nextLine();
+            if (currentLine.isBlank()) {
+                continue;
+            }
+            String taskType = currentLine.split(" \\| ")[0];
+            String taskDescription = currentLine.split(" \\| ")[2];
+            String isTaskDone = currentLine.split(" \\| ")[1];
+            switch (taskType) {
+            case "T":
+                Task toDoTask = new ToDo(taskDescription);
+                if (isTaskDone.equals("1")) {
+                    toDoTask.taskDone();
+                }
+                tasks.add(toDoTask);
+                break;
+            case "D":
+                String deadlineBy = currentLine.split(" \\| ")[3];
+                Task deadlineTask = new Deadline(taskDescription, deadlineBy);
+                if (isTaskDone.equals("1")) {
+                    deadlineTask.taskDone();
+                }
+                tasks.add(deadlineTask);
+                break;
+            case "E":
+                String eventAt = currentLine.split(" \\| ")[3];
+                Task eventTask = new Events(taskDescription, eventAt);
+                if (isTaskDone.equals("1")) {
+                    eventTask.taskDone();
+                }
+                tasks.add(eventTask);
+                break;
+            default:
+                System.out.println("Undefined task type - should not happen");
+                break;
+            }
+        }
+    }
+    private static void saveTaskList(ArrayList<Task> tasks, String dataListFilePath) throws IOException {
+        FileWriter fw = new FileWriter(dataListFilePath);
+        //for (Task task : t)
```

Comment:
Do not retain commented out code. If necessary, you can retrieve it from Git later.